src/containers/GeneralInfo/GeneralInfo.tsx

```diff
@@ -30,6 +31,13 @@ export const GeneralInfo = ({
         setSelectedNodeIndex(index);
     };
 
+    const compareNodeVersions = (): boolean => {
+        if (node.latestNodeVersion && node.nodeInfo && node.nodeInfo.version) {
```

Comment:
you can use here as well "Loading" concept so you don't have to check in every part if `nodeInfo` exists.

Also, the condition if `node.latestNodeVersion` is not loaded should not report to user that there is update available. Rather it should have neutral status (because we don't know if node is up to date or not)